src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +565,222 @@ subroutine DustEmission (bounds, &
+
+            ! mean lowpass-filtered wind speed at 0.1 m saltation height (assuming aerodynamic roughness length = 1e-4 m globally for ease; also assuming neutral condition)
+            u_mean_slt(p) = (wnd_frc_slt/k) * log(0.1_r8 / 1e-4_r8)  ! translating from ustar (velocity scale) to actual wind
+
+            ! sd of lowpass-filtered wind speed
+            !if (obul(p)==0) then
+            !   zetaobu = 0
+            !else 
+               !zetaobu = zii(p) / obul(p)   ! For now zii is a constant of 1000 m in CLM -dml, 24 Aug 2021
+            !   zetaobu = 1000_r8 / obul(p)   ! For now zii is a constant of 1000 m in CLM -dml, 24 Aug 2021
+            !end if
+            !stblty(p) = zetaobu    ! zetaobu get outputted as the Obukhov stability parameter
+            stblty(p) = 0   ! -dmleung 2 Dec 2021: use stability = 0 for now, assuming no buoyancy contribution. Might uncomment the above lines in future revisions.
+            if ((12_r8 - 0.5_r8 * stblty(p)) .GE. 0.001_r8) then ! should have used 0 theoretically; used 0.001 here to avoid undefined values
+               u_sd_slt(p) = wnd_frc_slt * (12_r8 - 0.5_r8 * stblty(p))**0.333_r8
+            else
+               u_sd_slt(p) = 0.001_r8   ! should have used 0 theoretically; used 0.001 here to avoid undefined values
             end if
 
-            ! the following comes from subr. flx_mss_vrt_dst_ttl_MaB95_get
-            ! purpose: diagnose total vertical mass flux of dust from vertically
-            !          integrated streamwise mass flux
+            ! threshold velocities
+            ! Here wnd_frc_thr_slt is the fluid threshold; wnd_frc_thr_dry(p) is the dry fluid threshold; B_it*wnd_frc_thr_dry(p) is the impact threshold
+            ! fluid threshold wind at 0.1 m saltation height
+            u_fld_thr(p) = (wnd_frc_thr_slt/k) * log(0.1_r8 / 1e-4_r8)
+            ! impact threshold wind at 0.1 m saltation height
+            u_impct_thr(p) = (wnd_frc_thr_slt_it/k) * log(0.1_r8 / 1e-4_r8)  ! to avoid model error
+
+            ! threshold crossing rate
+            numer = (u_fld_thr(p)**2_r8 - u_impct_thr(p)**2_r8 - 2_r8 * u_mean_slt(p) * (u_fld_thr(p) - u_impct_thr(p)))
+            denom = (2_r8 * u_sd_slt(p)**2_r8)
+            ! Truncate to zero if the expression inside exp is becoming too large
+            if ( numer/denom < 30._r8 )then
+               thr_crs_rate(p) = (exp((u_fld_thr(p)**2_r8 - u_impct_thr(p)**2_r8 - 2_r8 * u_mean_slt(p) * (u_fld_thr(p) - u_impct_thr(p))) / (2_r8 * u_sd_slt(p)**2_r8)) + 1_r8)**(-1_r8)
+            else
+               thr_crs_rate(p) = 0.0_r8
+            end if
+
+            ! probability that lowpass-filtered wind speed does not exceed u_ft
+            prb_crs_fld_thr(p) = 0.5_r8 * (1_r8 + erf((u_fld_thr(p) - u_mean_slt(p)) / (1.414_r8 * u_sd_slt(p))))
```

Comment:
Another magic number 1.414

Reply:
We can change it to sqrt(2).

Reply:
Make sure it's sqrt(2.0_r8) so you get it in double precision

Reply:
Done.